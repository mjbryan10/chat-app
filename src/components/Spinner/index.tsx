import React from 'react';
import './styles.css';
//TODO: Add spinner styles and functionality.

/**
* Usage under MIT lisence
* @see https://epic-spinners.epicmax.co/
* @see https://github.com/epicmaxco/epic-spinners
*/

/**
 * A Loading spinner component
 */
const Spinner = () => {
   return (
      <div className="container">
         <div className="self-building-square-spinner">
            <div className="square"></div>
            <div className="square"></div>
            <div className="square"></div>
            <div className="square clear"></div>
            <div className="square"></div>
            <div className="square"></div>
            <div className="square clear"></div>
            <div className="square"></div>
            <div className="square"></div>
         </div>
      </div>
   );
};

export default Spinner;
