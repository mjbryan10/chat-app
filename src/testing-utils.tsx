import React from 'react';
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { dark } from './shared/theme';

const AllTheProviders = ({ children }: any) => {
  return (
    <ThemeProvider theme={dark}>
        {children}
    </ThemeProvider>
  )
  
}

const customRender = (ui?: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }