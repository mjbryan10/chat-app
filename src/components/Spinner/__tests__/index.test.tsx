import React from 'react';
import { render } from 'testing-utils';
import Spinner from '..';

describe('Spinner Component', () => {
  test('should render correctly', () => {
    const { asFragment } = render(<Spinner />);
 
    expect(asFragment()).toMatchSnapshot();
 });
  
})
