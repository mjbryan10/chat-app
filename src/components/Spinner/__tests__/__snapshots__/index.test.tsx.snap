// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="self-building-square-spinner"
    >
      <div
        class="square"
      />
      <div
        class="square"
      />
      <div
        class="square"
      />
      <div
        class="square clear"
      />
      <div
        class="square"
      />
      <div
        class="square"
      />
      <div
        class="square clear"
      />
      <div
        class="square"
      />
      <div
        class="square"
      />
    </div>
  </div>
</DocumentFragment>
`;
